src/routes/api/ticket/assign/+server.ts

```diff
@@ -0,0 +1,129 @@
+import {
+    AssignAgentToTicketResult,
+    assignAgentToTicket,
+    canAssignSelfToTicket,
+    getUserFromSession,
+    isAssignedAgent,
+    isAssignedDepartment,
+    isHeadSession,
+    removeTicketAgent,
+} from '$lib/server/database';
+import { AssertionError } from 'node:assert/strict';
+import type { RequestHandler } from './$types';
+import { StatusCodes } from 'http-status-codes';
+import { error } from '@sveltejs/kit';
+
+function resultToCode(result: AssignAgentToTicketResult) {
+    switch (result) {
+        case AssignAgentToTicketResult.Success:
+            return StatusCodes.CREATED;
+        case AssignAgentToTicketResult.NoTicket:
+        case AssignAgentToTicketResult.NoAgent:
+            return StatusCodes.NOT_FOUND;
+        default:
+            throw new AssertionError();
+    }
+}
+
+// eslint-disable-next-line func-style
+export const POST: RequestHandler = async ({ cookies, request }) => {
+    const form = await request.formData();
+
+    const tid = form.get('ticket');
+    if (tid === null || tid instanceof File) throw error(StatusCodes.BAD_REQUEST);
+
+    const dept = form.get('dept');
+    if (dept === null || dept instanceof File) throw error(StatusCodes.BAD_REQUEST);
+    const did = parseInt(dept, 10) >> 0;
+
+    const uid = form.get('uid');
```

Comment:
For consistency, I'd prefer saying `user` here instead.

> [!IMPORTANT]
> The corresponding `URLSearchParameters` in the client wrapper must also be updated.

```suggestion
    const uid = form.get('user');
```

Reply:
I'll fix this but I will say this is also present in other endpoints as well which would have to be fixed.